examples/device/board_test/Makefile

```diff
@@ -9,4 +9,10 @@ INC += \
 EXAMPLE_SOURCE += $(wildcard src/*.c)
 SRC_C += $(addprefix $(CURRENT_PATH)/, $(EXAMPLE_SOURCE))
 
+# board_test example is special example that doesn't enable device or host stack
+# This can cause some TinyUSB API missing, this hack to allow us to fill those API
+# to pass the compilation process
+CFLAGS +=	\
+	-D"tud_irq_handler(x)= " \
```

Comment:
Why does this cause symbols to be missing? It's not as nice-looking but I feel like an internal include guard (`$*INTERNAL_PREFIX)_IN_BOARD_TEST_EXAMPLE`) would indicate intent better.

Reply:
> So just so I understand... before this PR, TinyUSB had four special symbols: `tud_isr` and `tuh_isr`, `dcd_isr`, and `hcd_isr` (with [defines](https://github.com/hathach/tinyusb/commit/bbec47b647ab7b2e38644ffff60f45b54b791d42#diff-82aaa3cc625cca0e786868f5c6660f0cR51) relating the first two to the latter two).
> 
> Before this PR, `tud_isr` and `tuh_isr` were only used if a microcontroller supported _both_ host and device mode. If the microcontroller only supported device mode, `tud_isr` was ignored and we directly implemented the IRQ handler in the `dcd` implementations.

before this PR, tud_isr and tuh_isr still are the same as this PR, I just start doing this with port that is added later on after discussion on this matter in an issue

> As of this PR:
> 
> * `tuh_isr`/`hcd_isr` are left alone.

Yes, I will rename it when reworking with host. I am a bit lazy to pull out an host board for physical testing now. They should be renamed to _irq_hanlder() as well though.

> * `dcd_isr` was renamed to `dcd_irq_handler`.
> * `tud_isr` was renamed to `tud_irq_handler`.

Yes

> * `dcd files no longer implement the IRQ handler directly, even if they are device-only. It is the board file's responsibility to called `tud_irq_handler`.
> 
> Is this accurate?

Yes, it has another benefit as well, some mcus has the same IRQ Handler for host and device, when both enabled, it could just still work if application call both tud/tuh irq_handler

Reply:
> Why does this cause symbols to be missing? It's not as nice-looking but I feel like an internal include guard (`$*INTERNAL_PREFIX)_IN_BOARD_TEST_EXAMPLE`) would indicate intent better.

The board_test example has `CFG_TUSB_RHPORT0_MODE = OPT_MODE_NONE` which cause the dcd_irq_handler() not compiled at all, it throw out lot errors. Instead of having macro to check if device is enabled for each bsp ( which is not really needed when application use the stack). I just hack it into command line define.